resources/sdk/purecloudjavascript/templates/ApiClient.mustache

```diff
@@ -727,122 +866,134 @@ class ApiClient {
 	 * @returns {Promise} A Promise object.
 	 */
 	callApi(path, httpMethod, pathParams, queryParams, headerParams, formParams, bodyParam, authNames, contentTypes, accepts) {
-		var url = this.buildUrl(path, pathParams);
-		var request = superagent(httpMethod, url);
-
-		if (this.proxy && request.proxy) {
-			request.proxy(this.proxy);
-		}
+		return new Promise((resolve, reject) => {
```

Comment:
Can you summarize the changes to the `callApi` function? It looks like maybe you rearranged some parts to properly handle refreshing the token, but the diff is making it look like there were a lot of changes.

Reply:
For the purpose of calling the function recursively after refreshing the access token I created a nested function named "sendRequest" inside callApi.
sendRequest retains most of the functionality of callApi, references to "this" have been changed to "that" which is partially why the diff looks so different, the other reason being everything is nested a further level now.
The functionality for handling an expired access token has been added at line 970. Previously it called "reject(data);" when an error occurred, now it calls _handleExpiredAccessToken and recursively calls itself if a 401 code is received and shouldRefreshAccessToken is set. It will call "reject(data);" if the status code isn't 401, if shouldRefreshAccessToken is not set or an error occurred in _handleExpiredAccessToken.